hawc/services/epa/hero.py

```diff
@@ -29,20 +29,37 @@ def _force_int(val, default=None) -> int | None:
 
 
 def parse_article(content: dict) -> dict:
-    authors = normalize_authors(content.get("AUTHORS", "").split("; "))
-    authors_short = get_author_short_text(authors)
-    return dict(
-        json=content,
-        HEROID=_force_int(_parse_pseudo_json(content, "REFERENCE_ID")),
-        PMID=_force_int(_parse_pseudo_json(content, "PMID")),
-        doi=try_get_doi(_parse_pseudo_json(content, "doi")),
-        title=_parse_pseudo_json(content, "TITLE"),
-        abstract=_parse_pseudo_json(content, "ABSTRACT"),
-        source=_parse_pseudo_json(content, "SOURCE"),
-        year=_force_int(_parse_pseudo_json(content, "YEAR")),
-        authors=authors,
-        authors_short=authors_short,
-    )
+    if settings.HAWC_FEATURES.ENABLE_NEW_HERO:
+        authors = normalize_authors(content.get("authors", []))
+        return dict(
+            json=content,
+            HEROID=_force_int(content.get("id")),
+            PMID=_force_int(content.get("accession_number")),
+            doi=try_get_doi(content.get("doi", "")),
+            title=content.get("title"),
+            abstract=content.get("abstract"),
+            source=content.get(
+                "type_of_reference"
+            ),  # TODO: need to construct source from journal/issue/pages/etc.
```

Comment:
The new API doesn't return the source like the old one does, but it does return the journal title/issue #/pages separately. However I don't see anywhere that we actually use the source from HEROFetch, should I just leave it like this?

Reply:
thanks for the heads up; I'll take a look when I do the detailed review. Is this one complete from your end?